### DISCUSSION TITLE
Atom Family with reusable Table component

### DISCUSSION THREAD

**Discussion by k-1ee** [pmndrs/jotai#2041] [General]
I'm using atoms to address a common scenario in which an entire table component re-renders when toggling a checkbox for a particular row. 

I have "selected" row IDs stored as an array in an atom, and each table row component is wired up to get/set toggle state by using `useAtom` while providing the row ID to subscribe to. This works great when you wire up the table and its descendants to use a single atom. However, I want this table to be reusable in that the selection state (derived from an atom) can be unique across multiple instances of this table.

The implementation for this seems open-ended and I'd like to discuss best practice. What I'd like to avoid is passing down an atom as props to the table in order to black-box the coupling with Jotai as much as possible. 

Currently, I'm using atom-family like so:
- pass tableID as param to atom-family 
- pass tableID to table component in order to get/set correct atom

This allows me to render the same table component three times and have the selection state be unique across each rendered instance. I can easily read a particular table's selection state by using the tableID. 

The issue with this (and atom-family in general) is memory leak, and having to explicitly remove atoms in the family. For my case, this can be a problem as the # of mapped atoms in the family can grow quite large across different use-cases of the table component. 

With Jotai, what is the recommended approach to manage state within a reusable component such that state is: 1) unique to the rendered instance, and 2) consumable outside of said component?

**Comment by dai-shi** [comment]
> passing down an atom as props

Not an answer because the requirement is to avoid it, but that's what I think I did: https://blog.axlight.com/posts/learning-react-state-manager-jotai-with-7guis-tasks/

**Reply by k-1ee** [reply]
@dai-shi - seems you're referring to "Task 5: CRUD" in the blog? 

If there isn't a blaringly obvious convention for this scenario, I may have to go with the atoms-as-props route. Wanted to make sure I wasn't missing anything before committing to the atoms-as-props pattern.

**Reply by dai-shi** [reply]
Well, passing ids as props isn't a bad idea and, if the requirement is the reusability (independent from Jotai), it's good.
So, I'm not suggesting to go with the route. I actually look forward to your exploration with ids as props. The issue is, as you mention, possible memory leaks with `atomFamily`.
